src/screens/ProfileScreen/index.js

```diff
@@ -587,6 +581,16 @@ const ProfileScreen = ({route}) => {
     fetchMyProfile(true);
   }
 
+  const onHeaderOptionClosed = () => {
+    setSelectedPostForOption(null);
+    setIsOptionModalOpen(false);
+  };
+
+  const onHeaderOptionClicked = (post) => {
+    setSelectedPostForOption(post);
+    setIsOptionModalOpen(true);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Two new functions `onHeaderOptionClosed` and `onHeaderOptionClicked` are introduced. Ensure they are correctly bound (if necessary) and called at the appropriate times.
```diff
+ const onHeaderOptionClosed = () => {
+   setSelectedPostForOption(null);
+   setIsOptionModalOpen(false);
+ };
+
+ const onHeaderOptionClicked = (post) => {
+   setSelectedPostForOption(post);
+   setIsOptionModalOpen(true);
+ };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->